packages/@aws-cdk/aws-stepfunctions-tasks/lib/evaluate-expression.ts

```diff
@@ -56,18 +56,17 @@ export class EvaluateExpression implements sfn.IStepFunctionsTask {
   public bind(task: sfn.Task): sfn.StepFunctionsTaskConfig {
     const matches = this.props.expression.match(/\$[.\[][.a-zA-Z[\]0-9]+/g);
 
-    if (!matches) {
-      throw new Error('No paths found in expression');
+    let expressionAttributeValues: {} | undefined;
+    if (matches) {
+      expressionAttributeValues = matches.reduce(
```

Comment:
This expression is a very complex, to me. Defining a helper function, doesn't it just read a lot better to say something like:

```ts
const matches = ...;
const expressionAttributeValues = fromEntries(matches.map(m => [m, sfn.Data.stringAt(m)]));
```
?

Obviously, before Node 12, this requires the helper:

```ts
function fromEntries<A>(entries: Array<[string, A]>): Record<string, A> {
  const ret: Record<string, A> = {};
  for (const [k, v] of entries)) {
    ret[k] = v;
  }
  return ret;
}
```

Not a blocker, I just generally prefer defining helper functions to make the main expression more readable...